common/core/src/promise_utils.ts

```diff
@@ -69,6 +69,8 @@ export type HttpResponseCallback<TResult> = TripleValueCallback<TResult, any>;
  * const param = 42;
  * callbackToPromise((_callback) => foo(param, _callback)).then(result => { console.log(result); }, error => { console.error(error); });
  */
+export function callbackToPromise<TResult>(callBackOperation: (callback: Callback<TResult>) => void, userCallback: Callback<TResult>): void;
+export function callbackToPromise<TResult>(callBackOperation: (callback: Callback<TResult>) => void): Promise<TResult>;
```

Comment:
Not sure if here and in `errorCallbackToPromise` the overload is necessary. Those methods should not be called by customers and the callback parameter should determine if callback or Promise should be used. Does lack of overloads here break anything?